pkg/varsub/varsub_test.go

```diff
@@ -148,7 +154,7 @@ func TestSubstitute(t *testing.T) {
 		{
 			name:  "simple text with escaped empty white signs 2",
 			value: "Foo ${ %\n \r%\n} bar",
-			want:  "Foo ${\n \r} bar",
+			want:  "Foo ${ %\n \r%\n} bar",
 		},
```

Comment:
Edge-case: `${lorem} ${%lorem%} ${lorem}`    ->    `ipsum ipsum ${lorem}`
Should be `ipsum ${lorem} ipsum`.

I.e, the test proposed below breaks, result is
```
expected: "Foo ipsum ${lorem} ipsum"
actual  : "Foo ipsum ipsum ${lorem}"
```

```suggestion
		},
		{
			name:  "unescaped variables don't mess with substitution of matching vars after",
			value: "Foo ${lorem} ${%lorem%} ${lorem}",
			want:  "Foo ipsum ${lorem} ipsum",
		},
```
Only happens if the result of the unescape matches one of the vars after it.
I.e., this is passes:
```golang
{
	name:  "unescaped variables don't mess with substitution of non-matching vars after",
	value: "Foo ${lorem} ${%foo-bar%} ${lorem}",
	want:  "Foo ipsum ${foo-bar} ipsum",
},
```

I'd be kind of okay with not handling this case, but it would be very nice to fix now so it doesn't bite us in the butt 😅